### ISSUE TITLE
Exercise 5: Equals Error

### ISSUE THREAD

**Issue by GarlicButterSteak** [nus-cs2030/2223-s1#266]
Hi everyone, was wondering if anyone had come across this error before:

![Bug](https://user-images.githubusercontent.com/77563792/199719224-bd476399-ffe3-48e0-be14-43f723bb9246.JPG)

I initially thought that the error came from the possibility of passing a null into the method equals(Object obj), i.e., something like checking for equals(null), but this didn't seem to fix the issue when I included a conditional to check for both this and obj being nulls. I also thought that my Maybe constructors did not account for null but at first glance they seem like they do too hmmm

**Comment by junhuiii** [comment]
Hi I think you may want to consider the case where the current object that you are referring to is `null` and therefore does not have an `equals` method to do comparison with.

**Comment by wilsonwid** [comment]
I agree with @junhuiii -- you should have conditions that check whether both objects contained in the `Maybe` context are `null` and whether either one is `null`.

**Comment by izascia256** [comment]
Hi, the error is thrown as the in-built getters in the method are trying to retrieve a `null` value which throws a `NullPointerException`. You have to include some clauses that deal with `null` separately as mentioned by @junhuiii.

**Comment by ZYN2122** [comment]
Yep, you should check if the item is null first

**Comment by qitiancong** [comment]
you need to check if your current object is null.
